### ISSUE TITLE
Move to GTK+ 4.x

### ISSUE THREAD

**Issue by declension** [quodlibet/quodlibet#4057]
The proposed change
-------------------
* Require GTK+ 4.x (e.g. 4.4?)
* Use a couple of quick-win features to prove the point

Benefits of this change
-----------------------
In future:
* Get to use modern look and feel
* Get to use modern [GTK+ 4.0 widgets](https://docs.gtk.org/gtk4/visual_index.html)
* GL rendering back-end by default (I think?)
* `GtkListView` may allow us to significantly improve performance in a many browser views
* Also GLSL shaders (e.g. for #3445) 

Any issues this might cause
---------------------------
* It's been out for > 1.5 years, but "still kinda new"
* Can it even work at all for us?
* Excludes a fair few distros though [Ubuntu support seems fine back to 21.10](https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=gtk-4&searchon=names). How might Windows / MacOS fit here?
* etc

**Comment by declension** [comment]
OK the migration is pretty massive: https://docs.gtk.org/gtk4/migrating-3to4.html

**Comment by lazka** [comment]
Windows+MacOS should be fine afaik

**Comment by declension** [comment]
(worth a look: [LibHandy](https://gnome.pages.gitlab.gnome.org/libhandy/))

**Comment by Meriipu** [comment]
> Get to use modern look and feel

what does this even mean

**Comment by declension** [comment]
> > Get to use modern look and feel
> 
> what does this even mean?

Yes fair point, bit vague. I guess it's mainly around things brought by `libadwaita` (see [Gnome announcement](https://blogs.gnome.org/alexm/2021/12/31/libadwaita-1-0/) for details)

**Comment by brenthuisman** [comment]
What about a more cross-platform friendly toolkit like Qt?

**Comment by afontenot** [comment]
> * It's been out for > 1.5 years, but "still kinda new"

One issue is that KDE doesn't yet have a matching GTK 4 theme. Until this exists, a move from GTK 3 to GTK4 will mean that QL goes from native-looking to not native-looking across most QT platforms.

I have to say that personally I hope that the GTK 3 to 4 transition is not going to be the occasion of "modernizing" the UI too much. A visual refresh would be fine, but changing the overall program UX would be a mistake IMO. What we have now works pretty well for us.

> What about a more cross-platform friendly toolkit like Qt?

Rewriting all of the widget code in Qt would be basically impossible - no easier than rewriting the app from scratch. Also, Gtk already is cross-platform. Looks pretty decent on Windows. The stuff that doesn't fit is mostly custom widgets (which would have to be entirely rewritten with a switch to Qt) and icons (Windows doesn't have "icon themes" like Linux does, we'd need to find and ship appropriate icons for that platform regardless of the widget toolkit).
